app/Services/Gear/Reminders/GearRemindersService.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace App\Services\Gear\Reminders;
+
+use App\Mail\GearReminderNotificationMail;
+use App\Models\Gear;
+use App\Models\GearReminder;
+use App\Models\StravaActivity;
+use DB;
+use Illuminate\Support\Facades\Mail;
+
+class GearRemindersService
+{
+    public function store(
+        string $gearId,
+        string $name,
+        int $triggerAfterNumberOfActivities,
+        int $currentNumberOfActivities
+    ): void {
+        GearReminder::create([
+            'gear_id' => $gearId,
+            'name' => $name,
+            'trigger_after_number_of_activities' => $triggerAfterNumberOfActivities,
+            'current_number_of_activities' => $currentNumberOfActivities,
+        ]);
+    }
+
+    public function update(
+        GearReminder $gearReminder,
+        string $name,
+        int $triggerAfterNumberOfActivities,
+        int $currentNumberOfActivities
+    ): void {
+        $gearReminder->update([
+            'name' => $name,
+            'trigger_after_number_of_activities' => $triggerAfterNumberOfActivities,
+            'current_number_of_activities' => $currentNumberOfActivities,
+        ]);
+    }
+
+    public function destroy(GearReminder $gearReminder): void
+    {
+        $gearReminder->delete();
+    }
+
+    public function attachGearAndTriggerReminders(StravaActivity $stravaActivity): void
+    {
+        $nonDecommissionedGearWithReminders = Gear::with('reminders')
+            ->where(
+                'user_id',
+                $stravaActivity->rawActivity->stravaConnection->user_id
+            )->where(function ($query) {
+                $query->whereNull('decommissioned')
+                    ->orWhereDate('decommissioned', '>', now()->toDateString());
+            })
+            ->where('auto_attach_to_activities', true)
+            ->whereHas('reminders')
```

Comment:
This isn't quite right as gear that doesn't have reminders won't be auto-attached.
This should probably be split into two methods, one in `GearService` to attach gear, and then a modification of this method to handle only reminder updating / firing.